picasso/src/main/java/com/squareup/picasso/MediaStoreRequestHandler.java

```diff
@@ -103,6 +107,37 @@ static PicassoKind getPicassoKind(int targetWidth, int targetHeight) {
   }
 
   static int getExifOrientation(ContentResolver contentResolver, Uri uri) {
+    int exifOrientation = getExitOrientationFromFile(contentResolver, uri);
+    if (exifOrientation == ExifInterface.ORIENTATION_UNDEFINED) {
+      exifOrientation = getExifOrientationFromContentResolver(contentResolver, uri);
+    }
+    return exifOrientation;
+  }
+
+  static int getExitOrientationFromFile(ContentResolver contentResolver, Uri uri) {
```

Comment:
exif*. I think it would be more clear to parse the file path and pass it in here instead of a uri. It seems incorrect to call this "getExifOrientationFromFile" and not pass a file.